stream-chat-android-client/src/main/java/io/getstream/chat/android/client/utils/observable/Subscription.kt

```diff
@@ -1,11 +1,12 @@
 package io.getstream.chat.android.client.utils.observable
 
+import io.getstream.chat.android.client.ChatClient
 import io.getstream.chat.android.client.events.ChatEvent
 
 @Deprecated("Use the subscribe methods on ChatClient or ChannelController directly instead of events()")
 public class Subscription(
     private val observable: ChatObservable,
-    private var listener: ((ChatEvent) -> Unit)?,
+    private var listener: ChatClient.ChatEventListener<ChatEvent>?,
```

Comment:
This class is also deprecated already, no reason to update it

Reply:
> This is a good idea, but it's a breaking change for nearly all Java clients that currently use this API. Here's some examples I checked that work with the old API but won't compile with the new one:
> 
> ```java
> client.subscribe(new Function1<ChatEvent, Unit>() {
>     @Override
>     public Unit invoke(ChatEvent chatEvent) {
>         return null;
>     }
> });
> client.subscribe(new Function1<ChatEvent, Unit>() {
>     @Override
>     public Unit invoke(ChatEvent chatEvent) {
>         return Unit.INSTANCE;
>     }
> });
> client.subscribe(chatEvent -> null);
> client.subscribe(chatEvent -> Unit.INSTANCE);
> 
> client.subscribeForSingle(NewMessageEvent.class, new Function1<NewMessageEvent, Unit>() {
>     @Override
>     public Unit invoke(NewMessageEvent chatEvent) {
>         return Unit.INSTANCE;
>     }
> });
> client.subscribeForSingle(NewMessageEvent.class, chatEvent -> Unit.INSTANCE);
> ```
> 
> We should provide a nice migration path here if possible, or at least a very large and scary warning in the changelog.

What would a nice migration path look like? Supporting both the functional interface approach & the Kotlin function approach in subscribeFor methods, and deprecating one set?